x/common/types/key_rollapp_packet.go

```diff
@@ -40,15 +37,46 @@ func RollappPacketKey(rollappPacket *RollappPacket) []byte {
 
 	// Concatenate the byte slices directly.
 	result := append(statusPrefix, keySeparatorBytes...)
-	result = append(result, proofHeightBytes...)
-	result = append(result, keySeparatorBytes...)
 	result = append(result, rollappIdBytes...)
 	result = append(result, keySeparatorBytes...)
+	result = append(result, proofHeightBytes...)
+	result = append(result, keySeparatorBytes...)
 	result = append(result, packetUIDBytes...)
 
 	return result
 }
 
+func RollappPacketByStatusByRollappIDMaxProofHeightPrefixes(rollappID string, status Status, proofHeight uint64) ([]byte, []byte) {
```

Comment:
IMO this func is confusing here
it's logic can be done in the caller filter (`PendingByRollappIDByMaxHeight`)